xbmc/AutoSwitch.cpp

```diff
@@ -238,5 +238,5 @@ float CAutoSwitch::MetadataPercentage(const CFileItemList &vecItems)
     if(item->IsParentFolder())
       total--;
   }
-  return (float)count / total;
+  return (total != 0) ? ((float)count / total) : 0;
```

Comment:
a) Can a a vector with size n have n parentFolders()? 
b) Is it enough to early return if size == 0?